api/callbacks.go

```diff
@@ -342,11 +340,12 @@ func cHumanAddress(ptr *C.api_t, canon C.Buffer, human *C.Buffer) (ret C.GoResul
 		// we received an invalid pointer
 		return C.GoResult_BadArgument
 	}
-
 	api := (*GoAPI)(unsafe.Pointer(ptr))
 	c := receiveSlice(canon)
 	h, err := api.HumanAddress(c)
 	if err != nil {
+		// store the actual error message in the return buffer
+		*human = allocateRust([]byte(err.Error()))
```

Comment:
Can we rename this variable to something like `output`?

Reply:
I was reusing the same variables.

Last night, after shutting down the computer, I think I will just add another return value `err *Buffer` to all the calls to take any error result. Passing in another Buffer just requires us to allocate 24bytes of 0 and not any more memory. So no real drawback to add one more argument and it makes it all more clear.